web/pdf_viewer.js

```diff
@@ -44,7 +57,11 @@ class PDFViewer extends BaseViewer {
     if (numVisiblePages === 0) {
       return;
     }
-    this._resizeBuffer(numVisiblePages);
+    const visibleSet = new Set();
+    for (let i = 0; i < numVisiblePages; ++i) {
+      visibleSet.add(visiblePages[i].id);
+    }
+    this._resizeBuffer(numVisiblePages, visibleSet);
```

Comment:
Nit: Can you just pass `visiblePages` here and create the set in `resize`? I can see that you're using a set as anb optimization. Before this commit, the `pdf_viewer.js` script didn't need any knowledge of the buffer except for its size. Now it needs to know that the page ID is significant.

Reply:
I see four options here, none of them obviously best to me:
1. The way things are: the second parameter is optional, but using it involves the abstraction leak that you point out.
1. Replace just `visibleSet` with `visiblePages`. Now we're calling `_resizeBuffer` with two parameters, but `numVisiblePages` is derived from `visiblePages`, so it's a little redundant.
1. Replace both `numVisiblePages` and `visibleSet` with `visiblePages`. That fixes the redundancy, but now the other call site of `_resizeBuffer` has to be changed to also pass in an array of pages instead of just a number as it currently does. (That other call site is in `pdf_single_page_viewer.js`, which I was averse to touching since I didn't know how to test it.)
1. Replace both `numVisiblePages` and `visibleSet` with `visiblePages` at this call site, and inside `_resizeBuffer`, branch on whether the single argument is a number or an array, permitting the other call site to continue passing in just a number.

I'll do whatever you think is best.

Reply:
Option 2 seems to have the least issues. You could call the parameter `prioritizedPages` or something to make it more generic.